### ISSUE TITLE
Generic opening brace placement sniffs do not correctly support function return types

### ISSUE THREAD

**Issue by jrfnl** [squizlabs/PHP_CodeSniffer#1931]
The auto-fixer in the `Generic.Functions.OpeningFunctionBraceKernighanRitchie` sniff currently will break when a function has a return type declaration which is not on the same line as the closing parenthesis of the function declaration.

Example code:
```php
// This works fine:
function myFunction(): Something
{
    return null;
}

// This will create a fixer conflict:
function myFunction($a, $lot, $of, $params)
    : array
{
    return null;
}
```

**Comment by gsherwood** [comment]
I think the solution is to make sure this code doesn't produce an error:
```php
function myFunction($a, $lot, $of, $params)
    : array {
    return null;
}
```
Because the opening brace is technically on the same line as the declaration, but the declaration is split over multiple lines, which is the job of another sniff to check.

That will also make it consistent with this code, which also produces no error:
```php
function myFunction(
    $a,
    $lot,
    $of,
    $params
) {
    return null;
}
```

**Comment by gsherwood** [comment]
Should be fixed now. Thanks for reporting it.

**Comment by jrfnl** [comment]
@gsherwood Thanks for working on this.

Unfortunately, the fix has issues with trailing comments:
```php
function myFunction($a, $lot, $of, $params)
    : array // comment
{
    return null;
}

function myFunction($a, $lot, $of, $params)
    : array // phpcs:ignore Standard.Category.Sniff -- for reasons.
{
    return null;
}
```

When run against `Generic.Functions.OpeningFunctionBraceKernighanRitchie`, I would expect this to be fixed as (or an error thrown, but not auto-fixed):
```php
function myFunction($a, $lot, $of, $params)
    : array { // comment
    return null;
}

function myFunction($a, $lot, $of, $params)
    : array { // phpcs:ignore Standard.Category.Sniff -- for reasons.
    return null;
}
```

In reality, this will now be fixed (broken) as:
```php
function myFunction($a, $lot, $of, $params)
    : array {
 // comment
    return null;
}

function myFunction($a, $lot, $of, $params)
    : array {
 // phpcs:ignore Standard.Category.Sniff -- for reasons.
    return null;
}
```

**Comment by gsherwood** [comment]
@jrfnl That's not a broken fix - that's how this sniff fixes the brace position when there is a comment on the end. There's a unit test for it already (added here: https://github.com/squizlabs/PHP_CodeSniffer/commit/5c2a8a1e88a80697b5975f7429e0673ae3e5b250) although it's been fixing code like that for a lot longer.

**Comment by jrfnl** [comment]
@gsherwood For the "normal" comment I can understand this to be a valid fix, however for the new PHPCS annotations, it changes the meaning of the annotation completely, so I honestly do think something needs to be done about that.
